src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -74,17 +80,17 @@ class ApplyException(orchestrator.OrchestratorError):
     """
 
 
-def threaded(f):
-    def wrapper(*args, **kwargs):
+def threaded(f: FuncT) -> FuncT:
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
         t = threading.Thread(target=f, args=args, kwargs=kwargs)
         t.start()
         return t
 
-    return wrapper
+    return cast(FuncT, wrapper)
```

Comment:
i disagree. the `wrapper()` actually changes the return type of the wrapped function to `threading.Thread`.

Reply:
i think cannot type this decorator properly before https://github.com/python/mypy/issues/3157 is fixed.